pallets/credit/src/lib.rs

```diff
@@ -926,6 +927,28 @@ pub mod pallet {
             Self::credit_admin() == Some(user)
         }
 
+        // todo: check _caller's right in next pr
+        pub fn evm_update_credit(_caller: &H160, user: &T::AccountId, score: u64, add_flag: bool) {
+            if add_flag {
+                let credit_data = {
+                    match UserCredit::<T>::get(user) {
```

Comment:
how about using `add_or_update_credit`

Reply:
add_or_update_credit   modify saved staking credit, not need at this